drivers/hmis/app/graphql/types/hmis_schema/ce_referral.rb

```diff
@@ -57,20 +64,24 @@ def client
       load_ar_association(object, :client, scope: Hmis::Hud::Client.viewable_by(current_user))
     end
 
-    def current_step_name # There can be multiple steps currently in progress, but we're only going to show one in the project referrals table
-      instance = load_ar_association(object, :workflow_instance)
+    def current_steps
+      load_ar_association(object, :current_steps)
+    end
 
-      step_t = Hmis::WorkflowExecution::Step.arel_table
-      step_scope = Hmis::WorkflowExecution::Step.
-        where(status: ['available', 'in_progress']).
-        # Prefer to return a step that is in_progress over available
-        order(Arel::Nodes::Case.new.when(step_t[:status].eq('in_progress')).then(1).else(2)).
-        order(step_t[:id]) # Also sort by ID, to make sure this resolves deterministically
+    def current_step_time
+      # TODO(#7647) - use more accurate timestamp field rather than updated_at
+      load_ar_association(object, :current_steps).to_a.min_by(&:updated_at)&.updated_at
+    end
 
-      step = load_ar_association(instance, :steps, scope: step_scope).first
-      return if step.nil?
+    def updated_by
```

Comment:
This is just a sketch of an approach we might take here (but somewhat outside of spec). Let me know what you think!

Reply:
Thanks. I'm not sure if there is a meaningful distinction between the "updated_at" on ce_referrals vs the "updated_at"  on each step. If not, we could take the approach of updating `updated_at` on the referral table any time on of the sub tasks is touched? I also think we will want to add `last_updated_by_user_id` references (and likely created_by), because we will want to use those for reporting. That definitely feels out of scope here, though. I think you can leave it as-is or stubbed out, and add a reference to https://github.com/open-path/Green-River/issues/7678